src/main/java/com/revature/services/QuestionLogicImpl.java

```diff
@@ -21,55 +20,27 @@
 	
 	@Autowired
 	private QuestionRepository dao;
-	
-	@Autowired
-	private QuestionEvalRepository qEvalDao;
-	
-	@Autowired
-	private QuestionCommentRepository commentDao;
-	
-	@Autowired
-	private SubjectRepository subjectDao;
+
+    @Autowired
+    private JsonValidation validation;
+
+    @PersistenceContext
+    private EntityManager entityManager;
+
 	
 //CREATE-----------------------------------
 	@Override
 	@Transactional
 	public QuestionPool createQuestion(QuestionPool question) {
 		
-		if (question.getMaxCommunicationScore() == null) {
-			throw new ConstraintViolationException("Missing required field maxCommunicationScore (Integer)", null);
-		}
-		
-		if (question.getMaxKnowledgeScore() == null) {
-			throw new ConstraintViolationException("Missing required field maxKnowledgeScore (Integer)", null);
-		}
-		
-		if (question.getQuestionText() == null) {
-			throw new ConstraintViolationException("Missing required field questionText (String)", null);
-		}
-		
-		if (question.getSubject() == null || question.getSubject().getId() == null) {
-			throw new ConstraintViolationException("Missing required field subject.id (Integer)", null);
-		}
+		validation.validateQuestionPoolFields(question);
 		
 		question.setUseCount(0);
-		question = dao.save(question);
-		// retrieve subject text - not auto filled
-		question.getSubject().setSubject(subjectDao.findOne(question.getSubject().getId()).getSubject());
-		return question;
-	}
+        question.setDateLastUsed(null);
+		question = dao.saveAndFlush(question);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "question". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS1226)